lock/src/main/java/com/auth0/android/lock/utils/EnterpriseConnectionMatcher.java

```diff
@@ -0,0 +1,131 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.auth0.android.lock.utils;
+
+import android.support.annotation.Nullable;
+
+import com.auth0.Auth0Exception;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Helper class to find out which email domains can be valid for the current Auth0 configuration.
+ */
+public class EnterpriseConnectionMatcher {
+
+    private static final String DOMAIN_KEY = "domain";
+    private static final String DOMAIN_ALIASES_KEY = "domain_aliases";
+
+    private List<Strategy> strategies;
+
+    public EnterpriseConnectionMatcher(List<Strategy> strategies) {
+        this.strategies = new ArrayList<>();
+        if (strategies == null) {
+            return;
+        }
+
+        for (Strategy s : strategies) {
+            if (s.getType() == Strategies.Type.ENTERPRISE) {
+                this.strategies.add(s);
+            }
+        }
+    }
+
+    /**
+     * Tries to find a valid domain with the given input.
+     *
+     * @param email to search the Domain for.
+     * @return a Connection if found, null otherwise.
+     */
+    @Nullable
+    public Connection parse(String email) {
+        String domain = extractDomain(email);
+        if (domain.isEmpty()) {
+            return null;
+        }
+
+        domain = domain.toLowerCase();
+        for (Strategy s : strategies) {
+            for (Connection c : s.getConnections()) {
+                String mainDomain = c.getValueForKey(DOMAIN_KEY);
+                if (mainDomain != null && mainDomain.contains(domain)) {
+                    return c;
+                }
+
+                List<String> aliases = c.getValueForKey(DOMAIN_ALIASES_KEY);
+                if (aliases != null) {
+                    for (String d : aliases) {
+                        if (d.contains(domain)) {
+                            return c;
+                        }
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Extracts the domain part from the email
+     *
+     * @param email to parse
+     * @return the domain String if found, an empty String otherwise
+     */
+
+    public String extractDomain(String email) {
+        int indexAt = email.indexOf("@") + 1;
+        if (indexAt == 0) {
+            return "";
+        }
+        int indexDot = email.indexOf(".", indexAt);
+        String domain;
+        if (indexDot == -1) {
+            domain = email.substring(indexAt);
+        } else {
+            domain = email.substring(indexAt, indexDot);
+        }
+        if (domain.isEmpty()) {
+            return "";
+        }
+        return domain;
+    }
+
+    /**
+     * Extracts the username part from the email
+     *
+     * @param email to parse
+     * @return the username String if found, an empty String otherwise
+     */
+    public String extractUsername(String email) {
```

Comment:
@hzalaz this method is used on the `DomainFormView` class, when pre-filling the next username/email form with the right text (full email or only the username part)